docs/design/yarn-v1.md

```diff
@@ -0,0 +1,135 @@
+use the latest 1.x release for the prefetch, since 1.x is no longer under active development and we never previously allowed users to specify a specific
+Yarn version to process their request in Cachito. We could add more flexibility with a later enhancement if needed.
+
+Corepack could be configured to ignore user project configuration and use the global Yarn 1.x default by specifying the following
+[environment variable](https://github.com/nodejs/corepack#environment-variables):
+
+    COREPACK_ENABLE_PROJECT_SPEC=0
+
+Alternatively we could configure corepack explicitly using the packageManager field in package.json like is done in Cachi2's Yarn v2+ implementation.
+
+We will also need to set the following to prevent corepack from prompting for interactive user input when downloading Yarn:
+
+    COREPACK_ENABLE_DOWNLOAD_PROMPT=0
+
+### Project Configuration
+The [package.json](https://classic.yarnpkg.com/en/docs/package-json) and [yarn.lock](https://classic.yarnpkg.com/en/docs/yarn-lock)
+configuration files must be present in order to process the request.
+
+The node_modules directory should be absent. I'm not certain at the moment how to guard against a user-specified modulesFolder at build-time.
+
+Any additional configuration settings in the repository via [.yarnrc](https://classic.yarnpkg.com/en/docs/yarnrc) or .npmrc
+files should be ignored (at least initially) for the prefetch by specifying ```--no-default-rc``` for the yarn install command.
+We never previously honored user-defined settings with Cachito.
+
+### SBOM Generation
+The Yarn 1.x CLI doesn't provide enough information to generate an SBOM, so the yarn.lock and package.json files will need to
+be parsed in order to gather the necessary data. The code used to process these files will need to be
+[imported from or inspired by Cachito](https://github.com/containerbuildsystem/cachito/blob/63b8ec0ea615d114ccfa0d08dc0bec49e60e6a75/cachito/workers/pkg_managers/yarn.py)
+and refactored/improved.
+
+#### Key Points
+ - The yarn.lock file can be parsed with [pyarn](https://github.com/containerbuildsystem/pyarn) like in cachito
+ - Workspaces will need to be [processed separately](https://github.com/containerbuildsystem/cachito/blob/63b8ec0ea615d114ccfa0d08dc0bec49e60e6a75/cachito/workers/pkg_managers/yarn.py#L95) from package.json since they do not appear in the yarn.lock file
+ - Dev dependencies can be [determined similiarly](https://github.com/containerbuildsystem/cachito/blob/63b8ec0ea615d114ccfa0d08dc0bec49e60e6a75/cachito/workers/pkg_managers/yarn.py#L140) to cachito
+ - We will need to ensure that there are no collisions in the names of the archive files being added to the offline mirror
+
+
+### Avoiding Arbitrary Code Execution
+Yarn 1.x appears to honor the ```--ignore-scripts``` option for the install command. It can also be configured via environment variable.
+Used https://github.com/chmeliik/js-lifecycle-scripts/tree/yarn to verify.
```

Comment:
Does `--ignore-scripts` also apply to the scripts of your git dependencies?

(i.e. something similar to https://github.com/chmeliik/js-lifecycle-scripts/commit/f4269945d491c71bd19000883569a4b92f2785aa, but change the dep from a registry dep to a git dep)

Reply:
It does appear to. I had added that project (specifically the yarn branch) as a git dependency for another project during testing and didn't observe the scripts execute :man_shrugging:

Reply:
\o/